src/php/ext/grpc/channel.c

```diff
@@ -162,6 +166,72 @@ void generate_sha1_str(char *sha1str, char *str, php_grpc_int len) {
   make_sha1_digest(sha1str, digest);
 }
 
+bool php_grpc_persistent_list_delete_unused_channel(char* target, target_bound_le_t* target_bound_status){
+  char *key = NULL;
+  zval *data;
+  int key_type;
+  PHP_GRPC_HASH_FOREACH_STR_KEY_VAL_START(&grpc_persistent_list, key, key_type, data)
+    if (key_type != HASH_KEY_IS_STRING || key == NULL) {
```

Comment:
where key_type is assigned?

Reply:
In PHP5: [PHP_GRPC_HASH_FOREACH_STR_KEY_VAL_START](https://github.com/grpc/grpc/blob/master/src/php/ext/grpc/php7_wrapper.h#L86)

In PHP7: [PHP_GRPC_HASH_FOREACH_STR_KEY_VAL_START](https://github.com/grpc/grpc/blob/master/src/php/ext/grpc/php7_wrapper.h#L86:9)